src/iterator.rs

```diff
@@ -37,26 +38,63 @@ impl<'a> PointQuery<'a> {
     }
 }
 
+#[derive(Clone, Debug, Default)]
+pub struct PointQuery<'a> {
+    pub attributes: Vec<&'a str>,
+    pub location: PointLocation,
+    pub filter: HashMap<&'a str, ClosedInterval<f64>>,
+}
+
 /// Iterator over the points of a point cloud node within the specified PointCulling
 /// Essentially a specialized version of the Filter iterator adapter
-pub struct FilteredIterator {
+pub struct FilteredIterator<'a> {
     pub culling: Box<dyn PointCulling<f64>>,
+    pub filter: &'a HashMap<&'a str, ClosedInterval<f64>>,
     pub node_iterator: NodeIterator,
 }
 
-impl Iterator for FilteredIterator {
+fn update_keep<T>(keep: &mut [bool], data: &[T], interval: &ClosedInterval<f64>)
+where
+    T: ToPrimitive,
+{
+    for (k, v) in keep.iter_mut().zip(data) {
+        if let Some(v) = v.to_f64() {
+            *k &= interval.contains(v);
+        }
+    }
+}
+
+impl<'a> Iterator for FilteredIterator<'a> {
     type Item = PointsBatch;
 
     fn next(&mut self) -> Option<PointsBatch> {
         let culling = &self.culling;
         self.node_iterator.next().map(|mut batch| {
-            let keep: Vec<bool> = batch
+            let mut keep: Vec<bool> = batch
                 .position
                 .iter()
                 .map(|pos| {
                     culling.contains(&<Point3<f64> as cgmath::EuclideanSpace>::from_vec(*pos))
                 })
                 .collect();
+            for (attrib, interval) in self.filter {
+                if let Some(data) = batch.attributes.get(*attrib) {
+                    match data {
+                        AttributeData::U8(d) => update_keep(&mut keep, d, interval),
```

Comment:
Use the `match_attr_data` macro?

Reply:
Difficult because of the `unimplemented!` vector types.